### PR TITLE
Change middleware action annotations

### LINKED ISSUE #47: Add before/after action filters
Add before/after action filters. If the filter is empty, we must intercept all action, otherwise intercept only the actions that is passed.

### PR DESCRIPTION
**Pull request by flaviokreis**
- [x] Added filter to before/after actions annotations

It is possible set filter empty, in this case the method intercept all actions.

It closes #47.

### CODE REVIEW

File: core/src/main/java/com/github/raulccabreu/redukt/middlewares/BaseAnnotatedMiddleware.kt [hunk 1]
```diff
@@ -53,15 +53,29 @@ abstract class BaseAnnotatedMiddleware<T> : Middleware<T> {
...
+
         verifyNumberOfArguments(method)
 
-        interceptBefores.add(method)
+        if (annotation.filter.isEmpty())
+            interceptBefores.add(method)
+        else
+            annotation.filter.forEach {
+                befores.put(it, method)
+            }
```
Comment (niltonvasques) [inline core/src/main/java/com/github/raulccabreu/redukt/middlewares/BaseAnnotatedMiddleware.kt]: A possible way to reduce duplication here.

```kotlin
private fun addBeforeActions(method: Method) {
  val annotation = method.getAnnotation(BeforeActions::class.java) as BeforeActions
  verifyNumberOfArguments(method)
  filtering(annotation, interceptBefores, befores, method)
}

private fun filtering(val annotation, val methods, val mapToMethod, val method) {
   if (annotation.filter.isEmpty()) methods.add(method)
   else annotation.filter.forEach { mapToMethod.put(it, method) }
}
```

**Review by niltonvasques — CHANGES_REQUESTED** [review]
Very nice improvements in the annotated actions.

File: core/src/main/java/com/github/raulccabreu/redukt/middlewares/BaseAnnotatedMiddleware.kt (hunk 1 above)
Comment (flaviokreis) [reply core/src/main/java/com/github/raulccabreu/redukt/middlewares/BaseAnnotatedMiddleware.kt]: good solution.

**Review by niltonvasques — APPROVED** [review]
Better than expected :white_check_mark:

**Pull request merged**